helix/components/plots.py

```diff
@@ -17,9 +17,59 @@ def plot_box(plot_dir: Path, box_title: str):
     if plot_dir.exists():
         plots = sorted(plot_dir.iterdir(), key=lambda x: x.stat().st_ctime)
         with st.expander(box_title, expanded=len(plots) > 0):
+            # Group plots by model and type
+            plot_groups = {}
             for p in plots:
-                if p.name.endswith(".png"):
-                    st.image(str(p))
+                if not p.name.endswith(".png"):
+                    continue
+
+                # Extract model name and plot type from filename
+                parts = p.stem.split("-")
+                if len(parts) >= 2:
+                    model_name = parts[0]
+                    if model_name not in plot_groups:
+                        plot_groups[model_name] = {
+                            "Train": [],
+                            "Test": [],
+                            "Coefficients": [],
+                            "Other": [],
+                        }
+
+                    if "Train" in parts:
+                        plot_groups[model_name]["Train"].append(p)
+                    elif "Test" in parts:
+                        plot_groups[model_name]["Test"].append(p)
+                    elif "coefficients" in parts:
+                        plot_groups[model_name]["Coefficients"].append(p)
+                    else:
+                        plot_groups[model_name]["Other"].append(p)
```

Comment:
To fix the complexity, you can take this part and make a separate function, and then call that function here.